syntax = "proto3";

// Rikka is the service for all AI related tasks.
service Rikka {
  rpc Query(QueryRequest) returns (AnswerReply) {}
  rpc Summarize(SummaryRequest) returns (SummaryReply) {}
  rpc GenerateQuestion(QuestionRequest) returns (QuestionReply) {}
}

message QueryRequest {
  // The passage to query upon. Maximum of 384 words, but subject to change and
  // updated information can be found in Rikka the GitHub settings file.
  string passage = 1;

  // The question to ask on the provided passage. Maximum of 64 words.
  string question = 2;
}

message AnswerReply {
  // The answer to the previously provided question.
  string answer = 1;

  // The probability of the answer returned by the neural network.
  float score = 2;
}

message SummaryRequest {
  // The passage to summarize.
  string passage = 1;
}

message SummaryReply {
  // The summarized version of the previously provided passage.
  string summary = 1;
}

message QuestionRequest {
  // The passage, of any size, to generate a question from.
  string passage = 1;

  // The keyword for the basis of the generated question.
  string term = 2;
}

message QuestionReply {
  // The question generated from the previously provided passage and term.
  string question = 1;
}
